spacemesh/v2alpha1/layer.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+import "spacemesh/v2alpha1/block.proto";
+
+package spacemesh.v2alpha1;
+
+message Layer {
+  oneof versioned {
+    BlockV1 v1 = 1;
+  }
+}
+
+message LayerV1 {
+  uint32 number = 1; // layer number - not hash - layer content may change
+  enum LayerStatus {
+    LAYER_STATUS_UNSPECIFIED = 0; // not yet approved or confirmed
+    LAYER_STATUS_APPROVED = 1; // approved by hare
+    LAYER_STATUS_CONFIRMED = 2; // confirmed by tortoise
+  }
+  LayerStatus status = 2;
+  bytes hash = 3; // computer layer hash - do we need this?
+  Block blocks = 4; // layer's block
+  bytes root_state_hash = 5; // the root state hash of global state in this layer. if unset block was not applied optimistically.
```

Comment:
i think this needs to be improved, i suggest following:
- consensus hash - for user to understand that they are on the right side of the fork (if such exists)
- state hash - fingeprint of the computed state at the layer
- cumulative state hash - cumulative fingerpint that uniquely identifies state since genesis
- block hash

layer should be reported when it was applied to the state.  i suggest to remove this unknown state. there should be a field to express that it was only applied because of hare, and not yet verified by tortoise . maybe `validity` which is equal to hare or tortoise

Reply:
What's the difference between state hash and "cumulative state hash" here?

Reply:
we don't do cumulative at the moment. the state hash is a hash of the changed accounts in a given layer.

https://github.com/spacemeshos/go-spacemesh/issues/5677